hrms/hr/utils.py

```diff
@@ -721,59 +723,69 @@ def get_matching_queries(
 	filter_by_reference_date=None,
 	from_reference_date=None,
 	to_reference_date=None,
+	common_filters=None,
 ):
 	"""Returns matching queries for Bank Reconciliation"""
 	queries = []
 	if transaction.withdrawal > 0:
 		if "expense_claim" in document_types:
 			ec_amount_matching = get_ec_matching_query(
-				bank_account, company, exact_match, from_date, to_date
+				bank_account, company, exact_match, from_date, to_date, common_filters
 			)
 			queries.extend([ec_amount_matching])
 
 	return queries
 
 
-def get_ec_matching_query(bank_account, company, exact_match, from_date=None, to_date=None):
+def get_ec_matching_query(
+	bank_account, company, exact_match, from_date=None, to_date=None, common_filters=None
+):
 	# get matching Expense Claim query
+	filters = []
+	ec = qb.DocType("Expense Claim")
+
 	mode_of_payments = [
 		x["parent"]
 		for x in frappe.db.get_all(
 			"Mode of Payment Account", filters={"default_account": bank_account}, fields=["parent"]
 		)
 	]
-
-	mode_of_payments = "('" + "', '".join(mode_of_payments) + "' )"
 	company_currency = get_company_currency(company)
 
-	filter_by_date = ""
+	filters.append(ec.docstatus == 1)
+	filters.append(ec.is_paid == 1)
+	filters.append(ec.clearance_date.isnull())
+	filters.append(ec.mode_of_payment.isin(mode_of_payments))
+	if exact_match:
+		filters.append(ec.total_sanctioned_amount == common_filters.amount)
+	else:
+		filters.append(ec.total_sanctioned_amount.gt(common_filters.amount))
```

Comment:
This doesn't work since the function allows common_filters to be None by default. Ref: https://github.com/alyf-de/banking/issues/108

@ruthra-kumar what is it supposed to do if common_filters=None

Reply:
In this case - get_ec_matching_query(); common_filter is only used for deciding whether to match the exact amount or not. So, if not present, you can just fetch Expense Claims without filter on the sanctioned amount.

Having said that, common_filter is purposefully passed to all of the matching queries with it being built manually https://github.com/frappe/erpnext/blob/621f0f343a49fa568bb3f81cf27b005fc40cebcf/erpnext/accounts/doctype/bank_reconciliation_tool/bank_reconciliation_tool.py#L507-L516. I wonder, how it was called without the filter? Was there any class override?

Reply:
In the banking app I linked above, it has its own implementation of `check_matching` for the bank reco tool beta, which calls `get_matching_queries` from the hooks for all installed apps.
There we did not tweak it **yet** to pass `common_filters`. Either way if `common_filters` defaults to `None`, this `filters.append(..` should check for its existence first. I'll